easybuild/tools/modules.py

```diff
@@ -643,7 +654,7 @@ def show(self, mod_name):
             ans = MODULE_SHOW_CACHE[key]
             self.log.debug("Found cached result for 'module show %s' with key '%s': %s", mod_name, key, ans)
         else:
-            ans = self.run_module('show', mod_name, check_output=False, return_output=True)
+            ans = self.run_module('show', mod_name, check_output=False, return_stderr=True)
```

Comment:
I don't think this is correct, since Lmod may be configured to spit out the output of `module show` to stdout (via `$LMOD_REDIRECT` set to `1` for example); that's how we have configured Lmod...

I also don't fully understand the motivation for this. You mentioned reducing of false positives, can you give an example of what you mean?

Reply:
But we don't call module show, do we? We call the underlying lmod and that outputs configuration stuff (like the current module table) to stdout and user facing stuff to stderr.

Basically: the state can contain something which make the match succeed. I'm currently on leave so just check the current output stored here and you'll see what I mean. 

Alternatively we could actually use "module show" but that is a bash function. So I don't think that's reasonable

Reply:
An example what is printed to stdout:

```
import os
os.environ["MODULEPATH"] = "/lustre/ssd/ws/s3248973-EasyBuild/easybuild-haswell/modules/all:/home/s3248973/.local/EasyBuildDev/modules:/sw/modules/scs5/classic:/sw/modules/scs5/vis:/sw/modules/scs5/tools:/sw/modules/scs5/toolchain:/sw/modules/scs5/system:/sw/modules/scs5/phys:/sw/modules/scs5/perf:/sw/modules/scs5/numlib:/sw/modules/scs5/mpi:/sw/modules/scs5/math:/sw/modules/scs5/lib:/sw/modules/scs5/lang:/sw/modules/scs5/geo:/sw/modules/scs5/devel:/sw/modules/scs5/debugger:/sw/modules/scs5/data:/sw/modules/scs5/compiler:/sw/modules/scs5/chem:/sw/modules/scs5/cae:/sw/modules/scs5/bio:/sw/modules/scs5/base:/sw/modules/taurus/environment";
os.environ["_ModuleTable001_"] = "X01vZHVsZVRhYmxlXz17WyJNVHZlcnNpb24iXT0zLFsiY19yZWJ1aWxkVGltZSJdPTcyMDAwMCxbImNfc2hvcnRUaW1lIl09MC40MzAxNzY5NzMzNDI5LGRlcHRoVD17fSxmYW1pbHk9e30sbVQ9e1siRWFzeUJ1aWxkLWRldmVsb3AiXT17WyJmbiJdPSIvaG9tZS9zMzI0ODk3My8ubG9jYWwvRWFzeUJ1aWxkRGV2L21vZHVsZXMvRWFzeUJ1aWxkLWRldmVsb3AiLFsiZnVsbE5hbWUiXT0iRWFzeUJ1aWxkLWRldmVsb3AiLFsibG9hZE9yZGVyIl09Mixwcm9wVD17fSxbInN0YWNrRGVwdGgiXT0wLFsic3RhdHVzIl09ImFjdGl2ZSIsWyJ1c2VyTmFtZSJdPSJFYXN5QnVpbGQtZGV2ZWxvcCIsfSxtb2RlbnY9e1siZm4iXT0iL3N3L21vZHVsZXMvdGF1cnVzL2Vudmlyb25tZW50L21v";
os.environ["_ModuleTable002_"] = "ZGVudi9zY3M1Lmx1YSIsWyJmdWxsTmFtZSJdPSJtb2RlbnYvc2NzNSIsWyJsb2FkT3JkZXIiXT0xLHByb3BUPXtsbW9kPXtbInN0aWNreSJdPTEsfSx9LFsic3RhY2tEZXB0aCJdPTAsWyJzdGF0dXMiXT0iYWN0aXZlIixbInVzZXJOYW1lIl09Im1vZGVudi9zY3M1Iix9LH0sbXBhdGhBPXsiL2x1c3RyZS9zc2Qvd3MvczMyNDg5NzMtRWFzeUJ1aWxkL2Vhc3lidWlsZC1oYXN3ZWxsL21vZHVsZXMvYWxsIiwiL2hvbWUvczMyNDg5NzMvLmxvY2FsL0Vhc3lCdWlsZERldi9tb2R1bGVzIiwiL3N3L21vZHVsZXMvc2NzNS9jbGFzc2ljIiwiL3N3L21vZHVsZXMvc2NzNS92aXMiLCIvc3cvbW9kdWxlcy9zY3M1L3Rvb2xzIiwiL3N3L21vZHVsZXMvc2NzNS90b29sY2hhaW4iLCIvc3cv";
os.environ["_ModuleTable003_"] = "bW9kdWxlcy9zY3M1L3N5c3RlbSIsIi9zdy9tb2R1bGVzL3NjczUvcGh5cyIsIi9zdy9tb2R1bGVzL3NjczUvcGVyZiIsIi9zdy9tb2R1bGVzL3NjczUvbnVtbGliIiwiL3N3L21vZHVsZXMvc2NzNS9tcGkiLCIvc3cvbW9kdWxlcy9zY3M1L21hdGgiLCIvc3cvbW9kdWxlcy9zY3M1L2xpYiIsIi9zdy9tb2R1bGVzL3NjczUvbGFuZyIsIi9zdy9tb2R1bGVzL3NjczUvZ2VvIiwiL3N3L21vZHVsZXMvc2NzNS9kZXZlbCIsIi9zdy9tb2R1bGVzL3NjczUvZGVidWdnZXIiLCIvc3cvbW9kdWxlcy9zY3M1L2RhdGEiLCIvc3cvbW9kdWxlcy9zY3M1L2NvbXBpbGVyIiwiL3N3L21vZHVsZXMvc2NzNS9jaGVtIiwiL3N3L21vZHVsZXMvc2NzNS9jYWUiLCIvc3cvbW9kdWxlcy9zY3M1L2Jp";
os.environ["_ModuleTable004_"] = "byIsIi9zdy9tb2R1bGVzL3NjczUvYmFzZSIsIi9zdy9tb2R1bGVzL3RhdXJ1cy9lbnZpcm9ubWVudCIsfSxbInN5c3RlbUJhc2VNUEFUSCJdPSIvc3cvbW9kdWxlcy90YXVydXMvZW52aXJvbm1lbnQiLH0=";
os.environ["_ModuleTable_Sz_"] = "4";
_mlstatus = True
```
We don't want any of this for this task, do we?

Reply:
Indeed, `$LMOD_CMD python show` still prints the output we're interested in to `stderr`, even if Lmod is configured to redirect to stdout (for example by setting `$LMOD_REDIRECT`).

Reply:
That is because `$LMOD_CMD` is the underlying command. `module` uses this, splits stdout/stderr and processes those differently. Then it may pass through stderr or redirect to stdout